theano/configdefaults.py

```diff
@@ -1647,7 +1649,7 @@ def short_platform(r=None, p=None):
 # Allow to have easily one compiledir per device.
 compiledir_format_dict['device'] = config.device
 compiledir_format_keys = ", ".join(sorted(compiledir_format_dict.keys()))
-default_compiledir_format = ("compiledir_%(short_platform)s-%(processor)s-"
+default_compiledir_format = ("compiledir_%(hostname)s-%(short_platform)s-%(processor)s-"
```

Comment:
@abergeron just desribed part of the problem. Also, normally, the march flags are added in the key of each compiled module. So if the difference is only there, then it should not be a problem. There is a few exception about that. This would need refactoring for a full complete fix.

So, if you continue to get this problem, tell us more about the exact case you have. There is other fix needed than this one and we can discuss them to find the right fix.